src/overlays/actors/ovl_Obj_Chan/z_obj_chan.c

```diff
@@ -31,49 +39,404 @@ const ActorInit Obj_Chan_InitVars = {
+        float sin = Math_SinS(temp_s1);
+        float cos = Math_CosS(temp_s1);
+        spDC.x = sin * 8.0f;
+        spDC.y = (Rand_ZeroOne() * 12.0f) + 2.0f;
+        spDC.z = cos * 8.0f;
+        spD0.x = spDC.x * 0.23f;
+        spD0.y = (Rand_ZeroOne() * 5.0f) + 2.5f;
+        spD0.z = spDC.z * 0.23f;
+        temp_f0 = Rand_ZeroOne();
+        if (temp_f0 < 0.2f) {
+            phi_s0 = 0x60;
+        } else if (temp_f0 < 0.6f) {
+            phi_s0 = 0x40;
+        } else {
+            phi_s0 = 0x20;
+        }
+        new_var2 = (spA4 * Rand_ZeroOne());
+        EffectSsKakera_Spawn(globalCtx, &spDC, &spD0, &this->actor.world.pos, -260, phi_s0, 20, 0, 0, spA8 + new_var2,
+                             0, 0, 50, -1, OBJECT_TSUBO, object_tsubo_DL_001960);
+    }
+    func_800BBFB0(globalCtx, &this->actor.world.pos, 30.0f, 2, 20, 50, true);
+    func_800BBFB0(globalCtx, &this->actor.world.pos, 30.0f, 2, 10, 80, true);
+}
+
+void ObjChan_Update(Actor* thisx, GlobalContext* globalCtx) {
+    ObjChan* this = THIS;
 
-extern ColliderCylinderInit D_80BBAB30;
-extern InitChainEntry D_80BBAB5C[];
+    this->actionFunc(this, globalCtx);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+}
 
-extern UNK_TYPE D_06000AF0;
-extern UNK_TYPE D_06001960;
-extern UNK_TYPE D_06002358;
+void ObjChan_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    ObjChan* this = THIS;
+    Gfx* dl_opa;
+    Gfx* dl_xlu;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Init.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    Matrix_RotateY(this->rotation, true);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Destroy.s")
+    dl_opa = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19U);
+    gSPMatrix(dl_opa, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(dl_opa + 1, object_obj_chan_DL_000AF0);
+    POLY_OPA_DISP = dl_opa + 2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9A1C.s")
+    dl_xlu = func_8012C2B4(POLY_XLU_DISP);
+    gSPMatrix(dl_xlu, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(dl_xlu + 1, object_obj_chan_DL_000A10);
+    POLY_XLU_DISP = dl_xlu + 2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9B40.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9C08.s")
+    Matrix_StatePush();
+    if (this->stateFlags & OBJCHAN_STATE_ON_FIRE) {
+        func_80BBA930(this, globalCtx);
+    }
+    Matrix_StatePop();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9F24.s")
+void ObjChan_DrawPot(Actor* thisx, GlobalContext* globalCtx) {
+    ObjChan* this = THIS;
+    s32 pad;
+    Gfx* dl;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BBA2FC.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    dl = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19U);
```

Comment:
```suggestion
    dl = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19);
```
You can remove the U from this since it is always going to be unsigned in this case